src/VerifyWebauthnController.js

```diff
@@ -0,0 +1,177 @@
+  'vendor/lib/q',
+  'util/FactorUtil',
+  'views/mfa-verify/HtmlErrorMessageView'
+],
+function (Okta, FormController, FormType, CryptoUtil, FooterSignout, Q, FactorUtil, HtmlErrorMessageView) {
+
+  var _ = Okta._;
+
+  return FormController.extend({
+    className: 'mfa-verify verify-u2f',
+    Model: {
+      props: {
+        rememberDevice: 'boolean'
+      },
+
+      initialize: function () {
+        var rememberDevice = FactorUtil.getRememberDeviceValue(this.appState);
+        // set the initial value for remember device (Cannot do this while defining the
+        // local property because this.settings would not be initialized there yet).
+        this.set('rememberDevice', rememberDevice);
+      },
+
+      save: function () {
+        this.trigger('request');
+
+        return this.doTransaction(function (transaction) {
+          var factor = _.findWhere(transaction.factors, {
+            factorType: 'webauthn',
+            provider: 'FIDO'
+          });
+
+          var self = this;
+          return factor.verify().then(function (transaction) {
+            var factorData = transaction.factor;
+            self.trigger('request');
+
+            var deferred = Q.defer();
+
+            var idBytes = CryptoUtil.strToBin(factorData.profile.credentialId);
+            var challengeBytes = CryptoUtil.strToBin(factorData.challenge.challenge);
+            var options = factorData.challenge;
+
+            options.allowCredentials = [{
+              type: "public-key",
+              id: idBytes
+            }];
+            options.challenge = challengeBytes;
+
+            navigator.credentials.get({ "publicKey": options }).then(function (assertion) {
+              var rememberDevice = !!self.get('rememberDevice');
+              return factor.verify({
+                clientData: CryptoUtil.binToStr(assertion.response.clientDataJSON),
+                authenticatorData: CryptoUtil.binToStr(assertion.response.authenticatorData),
+                signatureData: CryptoUtil.binToStr(assertion.response.signature),
+                rememberDevice: rememberDevice
+              })
+              .then(deferred.resolve);
+            }).catch(function (err) {
+              self.trigger('errors:clear');
+                deferred.reject({
+                  xhr: {responseJSON: {errorSummary: err}}
+                });
+            });
+
+            return deferred.promise;
+          });
+        });
+      }
+    },
+
+    Form: {
+      autoSave: true,
+      hasSavingState: false,
+      title: _.partial(Okta.loc, 'factor.u2f', 'login'),
+      className: 'verify-u2f-form',
+      noCancelButton: true,
+      save: _.partial(Okta.loc, 'verify.u2f.retry', 'login'),
+      noButtonBar: function () {
+        return navigator.credentials.create == null;
+      },
+      modelEvents: {
+        'request': '_startEnrollment',
+        'error': '_stopEnrollment'
+      },
+
+      formChildren: function () {
+        var result = [];
```

Comment:
nit: clearer to call this children so it's clearer in the downstream form what we're doing